src/js/inPageNav.js

```diff
@@ -4,54 +4,160 @@ class InPageNav {
 
 	constructor (InPageNavEl, opts) {
 		this.inPageNavEl = InPageNavEl;
-		this.opts = opts || {}; // no opts right now
+		this.opts = opts || InPageNav.getOptions(InPageNavEl);
+
+		// the thing to look for within the container. Defaults to h2
+		if (this.opts.headingsSelector) {
```

Comment:
Instead of all the conditionals to assign default values, would it be worth using destructuring to set them up: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Default_values_2 ?